crates/apollo-compiler/tests/validation/recursion.rs

```diff
@@ -347,3 +349,28 @@ type Query {
     "#]]
     .assert_eq(&errors.to_string());
 }
+
+#[test]
+fn handles_directive_with_nested_input_types() {
```

Comment:
I was fixing up the changelog for 1.30.0 and when looking over this PR's description I couldn't figure out whether this test was incorrect, or I don't understand the description. Isn't the point that we safe guard from recursive argument types in directive definitions? If so, shouldn't this test return an error?